dotCMS/src/main/java/com/dotmarketing/services/ContentletServices.java

```diff
@@ -163,6 +164,10 @@ private static void invalidate(Contentlet content, Identifier identifier, boolea
 		if(content.getStructure().getStructureType()==Structure.STRUCTURE_TYPE_HTMLPAGE) {
 		    PageServices.removePageFile(APILocator.getHTMLPageAssetAPI().fromContentlet(content), identifier, EDIT_MODE);
 		}
+		if(content != null && content.isVanityUrl()){
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)